server.js

```diff
@@ -1,33 +1,93 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+
+import booksData from "./data/books.json";
+
+// SAVING OTHER DATA FOR LATER DEVELOPMENT OF THE PROJECT
+// import goldenGlobesData from "./data/golden-globes.json";
 // import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
 // import netflixData from './data/netflix-titles.json'
 // import topMusicData from './data/top-music.json'
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
-//
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get("/", (req, res) => {
+  res.send("Hello world");
+});
+
+// BOOKS API --> booksData
+
+app.get("/books", (req, res) => {
+  const searchString = req.query.search;
+
+  let filteredBooks = booksData;
+
+  if (searchString) {
+    filteredBooks = filteredBooks.filter(item => {
+      let itemTitle = item.title.toString().toLowerCase();
+      let itemAuthor = item.authors.toString().toLowerCase();
+
+      return (
+        itemTitle.includes(searchString) || itemAuthor.includes(searchString)
+      );
+    });
+  }
+
+  res.json(filteredBooks);
+});
+
+app.get("/books/:id", (req, res) => {
+  const id = req.params.id;
+
+  let bookid = booksData.filter(item => item.bookID === +id);
+
+  res.json(bookid);
+});
+
+app.get("/books/authors/:author", (req, res) => {
+  const author = req.params.author;
+
+  let filteredBooks = booksData;
+
+  if (author) {
+    filteredBooks = filteredBooks.filter(item => {
+      let itemAuthor = item.authors.toString().toLowerCase();
+      return itemAuthor.includes(author);
+    });
+  }
+  res.json(filteredBooks);
+});
+
+app.get("/info", (req, res) => {
+  const searchRating = req.query.rating;
+  const searchPages = req.query.pages;
+
+  let info = booksData.filter(
+    item =>
+      item.average_rating === +searchRating || item.num_pages === +searchPages
+  );
+
+  res.json(info);
+});
+
+app.get("/books/pages/:pages", (req, res) => {
```

Comment:
It would be more RESTful to have this filter as part of the `/books` endpoint and use query parameters to pass a pages param. Like `/books?pages=123` to filter where the number of pages is "123"